BuddySave/BackupDirectoryProvider.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace BuddySave;
+
+public class BackupDirectoryProvider : IBackupDirectoryProvider
```

Comment:
I would like to have BackupDir configurable for the cloud and for local.

This way all backups would reside in user defined folders. Cloud backup path should be derived from Cloud path by the application.

This is because right now we're gonna be saving backups in game's directory? Maybe that's ok 🤔

Reply:
btw, backups wont be in game directory, but inside the BuddySave apps execute directory

Reply:
Oh good! It's fine then.

If you want to reference exe's dir, you should use AppDomain.CurrentDomain.BaseDirectory, otherwise the "Start-in" dir is used. It will be different from exe, if, for example, run from .bat.